DatadogRUM/Sources/DDRUMMonitor+Internal.swift

```diff
@@ -1,49 +0,0 @@
-/*
- * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
- * This product includes software developed at Datadog (https://www.datadoghq.com/).
- * Copyright 2019-Present Datadog, Inc.
- */
-
-import Foundation
-import DatadogInternal
-
-extension DDRUMMonitor: DatadogInternalInterface {}
```

Comment:
@maxep FYI, I couldn't use `DatadogInternalInterface` for `RUMMonitorProtocol`, because it only works with classes. Decorating protocols this way seems much harder if not impossible.

To cut corners, I reused `DatadogInternalInterface` name in `RUMMonitorProtocol+Internal.swift`. IMO this shouldn't be a blocker for now - we can revisit it later.

Reply:
Indeed, the `DatadogExtended` does not work on protocols sadly :( 

We can avoid adding a public interface to `DatadogRUM` module by doing this:
```swift
/// Extends `RUMMonitorProtocol` with additional methods designed for Datadog cross-platform SDKs.
public extension RUMMonitorProtocol {
    /// Grants access to an internal interface utilized only by Datadog cross-platform SDKs.
    /// **It is not meant for public use** and it might change without prior notice.
    var _internal: DatadogExtension<RUMMonitorProtocol> {
        DatadogExtension(self)
    }
}

extension DatadogExtension where ExtendedType: RUMMonitorProtocol {
  ...
}
```

Then each method of the extension has to cast the monitor:
```swift
extension DatadogExtension where ExtendedType: RUMMonitorProtocol {
    /// Adds RUM long task to current view.
    /// - Parameters:
    ///   - time: the time of this command in cross-platform SDK
    ///   - duration: duration of the long task
    ///   - attributes: attributes to process along with this call
    func addLongTask(
        at time: Date,
        duration: TimeInterval,
        attributes: [AttributeKey: AttributeValue] = [:]
    ) {
        guard let monitor = type as? Monitor else {
            return
        }

        let longTaskCommand = RUMAddLongTaskCommand(time: time, attributes: attributes, duration: duration)
        monitor.process(command: longTaskCommand)
    }
}
```

I think it's preferable this way than adding a new public interface 👍

Reply:
I'm afraid it's not that easy. This ☝️ compiles in `DatadogRUM`, but fails to compile when used:

```swift
// ExampleAppDelegate.swift

import DatadogRUM

RUMMonitor.shared()._internal.addLongTask(at: Date(), duration: 0, attributes: [:])

// error: type 'any RUMMonitorProtocol' cannot conform to 'RUMMonitorProtocol'
//        RUMMonitor.shared()._internal.addLongTask(at: Date(), duration: 0, attributes: [:])

// note: only concrete types such as structs, enums and classes can conform to protocols
//        RUMMonitor.shared()._internal.addLongTask(at: Date(), duration: 0, attributes: [:])
//                            ^

// note: required by referencing instance method 'addLongTask(at:duration:attributes:)' on // 'DatadogExtension' where 'ExtendedType' = 'any RUMMonitorProtocol'
// public extension DatadogExtension where ExtendedType: RUMMonitorProtocol {
//       ^
```

Reply:
damn... :(

Reply:
No big deal IMO - we can improve it later with no breaking change for cross-platform 👌.

Reply:
Yes, I start to think that the `DatadogExtension` pattern is not well suited for mutations, especially in configuration where we aim to use `struct`...